resources/js/Pages/Collections/Components/CollectionsVoteReceivedModal/CollectionsVoteReceivedModal.tsx

```diff
@@ -0,0 +1,63 @@
+import { useTranslation } from "react-i18next";
+import { ButtonLink } from "@/Components/Buttons/ButtonLink";
+import { Dialog } from "@/Components/Dialog";
+
+export const CollectionsVoteReceivedModal = ({
+    onClose,
+    collection,
+}: {
+    onClose: () => void;
+    collection?: {
+        name: string;
+        slug: string;
+    };
+}): JSX.Element => {
+    const { t } = useTranslation();
+
+    const twitterLink =
+        collection !== undefined
+            ? `https://twitter.com/intent/tweet?text=${encodeURIComponent(
+                  t("pages.collections.collection_of_the_month.vote_received_modal.x_text", {
+                      collection: collection.name,
+                  }),
+              )}&url=${encodeURIComponent(
```

Comment:
Would you consider using `encodeURI` over `encodeURIComponent` to encode the whole URI instead of encoding parts separately, that would simplify the thing.